### PR TITLE
Assets: Use central class for registering scripts and styles 

### LINKED ISSUE #7386: Assets: Use central class for registering scripts and styles
## Task Description

Right now asset registration (and enqueueing) is scattered across a few places. There's room for improvement there, and I think a central `Assets` class (instead of the existing Trait) would come in handy here. Inspired by the [`Assets`](https://github.com/google/site-kit-wp/blob/15bf4bde55b5b6cec126a207c367dd33aefe5d3b/includes/Core/Assets/Assets.php) class in Site Kit (but we don't need all the extra classes.

### Implementation Brief

* Create a new `Assets` class under the `Google\Web_Stories` namespace, extending `Service_Base`
* Add this service to the list of services in the `Plugin` class and `phpstorm.meta.php`
* The `Assets::register()` method _registers_ all assets used by the plugins.
* There's an `enqueue_asset( $handle )` method
* Each class that needs some assets uses `$this->assets->enqueue_asset()`
* Remove `Register_Font` in favor of `Assets`
* Move code from `Assets` trait to this class
* Move asset registration from `\Google\Web_Stories\Embed_Base::register()` to this class

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Copy assets trait into a class. Add some helper methods. Inject the asset classes in all the places that need it. 
Add unit tests. 

## To-do

- [x] Add unit tests

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7386

### CODE REVIEW

File: includes/Admin/Google_Fonts.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	}
+
+	/**
+	 * Provides the asset handle.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @return string
+	 */
+	public function get_handle() {
```
Comment (swissspidy) [inline includes/Admin/Google_Fonts.php]: I don't think this method is needed.

Any consumer could just use `($this->google_fonts )::HANDLE`

Comment (spacedmonkey) [reply includes/Admin/Google_Fonts.php]: I got a lint error in PHPStorm when I did something similar.

Comment (swissspidy) [reply includes/Admin/Google_Fonts.php]: `$this->google_fonts::HANDLE` will cause warnings due to using PHP7+ syntax. But `($this->google_fonts )::HANDLE` should work. Can we try that?

File: includes/Plugin.php [hunk 2]
```diff
@@ -172,14 +174,16 @@ protected function get_bindings() {
...
+			Assets::class,
 			Experiments::class,
 			Admin\Meta_Boxes::class,
 			Locale::class,
 			Admin\Activation_Flag::class,
 			Integrations\Site_Kit::class,
 			Analytics::class,
 			Decoder::class,
-			Register_Font::class,
+			Amp_Player_Assets::class,
```
Comment (swissspidy) [inline includes/Plugin.php]: Nit: let's call that class `AMP_Story_Player_Assets` to make it clear it's for `<amp-story-player>`

Comment (spacedmonkey) [reply includes/Plugin.php]: Fixed

File: includes/Admin/Google_Fonts.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Admin/Google_Fonts.php]: I tried it. But it still failed in 5.6. See https://github.com/google/web-stories-wp/runs/2676667181

If you reference [Class Constants ](https://www.php.net/manual/en/language.oop5.constants.php). Even the examples have getter methods..

File: includes/Renderer/Stories/Renderer.php [hunk 3]
```diff
@@ -142,12 +155,19 @@ abstract class Renderer implements RenderingInterface, Iterator {
...
 		$this->query           = $query;
 		$this->attributes      = $this->query->get_story_attributes();
 		$this->content_overlay = $this->attributes['show_title'] || $this->attributes['show_date'] || $this->attributes['show_author'] || $this->attributes['show_excerpt'];
+		// TODO, find a way to inject this a cleaner way.
+		$injector = Services::get_injector();
+		if ( ! method_exists( $injector, 'make' ) ) {
+			return;
+		}
+		$this->assets                  = $injector->make( Assets::class );
+		$this->amp_story_player_assets = $injector->make( AMP_Story_Player_Assets::class );
```
Comment (spacedmonkey) [inline includes/Renderer/Stories/Renderer.php:170]: This code was written before we had the injector. It is much larger reflector to inject these classes here. Using the injector in the one place, saves doing it lots of and lots of places.

File: includes/Admin/Google_Fonts.php (hunk 1 above)
Comment (swissspidy) [reply includes/Admin/Google_Fonts.php]: Apparently one needs to do stuff like this on PHP 5:

```php
$fonts = $this->google_fonts;
echo $fonts::HANDLE
```

Let's keep the method for now but revisit in #7597

Comment (spacedmonkey) [reply includes/Admin/Google_Fonts.php]: Agreed, this is a small detail, it should not hold up the PR.

File: includes/Admin/Google_Fonts.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	 * Registers the google font style.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param WP_Styles $wp_styles WP_Styles instance.
+	 *
+	 * @return void
+	 */
+	public function register_style( WP_Styles $wp_styles ) {
+		// PHPCS ignore reason: WP will strip multiple `family` args from the Google fonts URL while adding the version string,
```
Comment (swissspidy) [inline includes/Admin/Google_Fonts.php]: I don't see a `phpcs:ignore` line?

Comment (spacedmonkey) [inline includes/Admin/Google_Fonts.php]:
```suggestion
```

**Review by swissspidy — APPROVED** [review]

**Pull request merged**